tests/publiccloud/qam_test.pm

```diff
@@ -0,0 +1,41 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Collect/store bootup time from publiccloud instances.
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+
+use base 'consoletest';
+use registration;
+use warnings;
+use testapi;
+use strict;
+use utils;
+
+sub run {
+    my ($self, $args) = @_;
+    select_console 'root-console';
+
+    assert_script_run("hostname -f");
+    assert_script_run("uname -a");
+    register_product();
+    zypper_call("in traceroute");
+    assert_script_run("ls /etc");
+    assert_script_run("cat /etc/hosts");
+    assert_script_run("cat /etc/os-release");
+    assert_script_run("ip a s");
+    assert_script_run("ip -6 a s");
```

Comment:
but the next module will continue, and maybe just do it's stuff on the openqa-jump-host. So we maybe need some kind of sanity. But I also don't have a good idea currently, how to track such issue in a common way.

Reply:
Also when a test module fails a special console for uploading logs is activated but perhaps `$serialdev` is still `sshserial` every command fails.

Reply:
The idea of updating post fail hooks still doesn't work for me because the openQA always reload the state saved after the last milestone and so the tunnel running on `user-console` goes down...

Reply:
JFYI this is fixed by inserting something like snippet bellow into every affected test module.
```perl
sub test_flags {
    return {no_rollback => 1} if get_var('PUBLIC_CLOUD');
}
```